tests/integration/test_hedged_requests/test.py

```diff
@@ -414,7 +372,7 @@ def test_async_connect(started_cluster):
     )
 
     NODES["node"].query(
-        "SELECT hostName(), id FROM distributed_connect ORDER BY id LIMIT 1 SETTINGS prefer_localhost_replica = 0, connect_timeout_with_failover_ms=5000, async_query_sending_for_remote=0, max_threads=1"
+        "SELECT hostName(), id FROM distributed_connect ORDER BY id LIMIT 1 SETTINGS prefer_localhost_replica = 0, connect_timeout_with_failover_ms=5000, async_query_sending_for_remote=0, max_threads=1, max_distributed_connections=1"
```

Comment:
`max_threads=1` can not help, see https://github.com/ClickHouse/ClickHouse/pull/53504

Reply:
But does `max_distributed_connections` changes anything here? Unlikely (https://github.com/ClickHouse/ClickHouse/issues/53287#issuecomment-1677535239)

Reply:
I tested it and yes, after this change all connections were processed in single thread using fibers, instead of using separate thread with fibers (that is actually meaningless I guess) for each connection